src/inmanta/server/agentmanager.py

```diff
@@ -1365,9 +1365,11 @@ async def _make_agent_config(
 name={opt.db_name.get()}
 username={opt.db_username.get()}
 password={opt.db_password.get()}
+service_connection_pool_min_size={agent_cfg.agent_db_connection_pool_min_size.get()}
+service_connection_pool_max_size={agent_cfg.agent_db_connection_pool_max_size.get()}
+service_connection_timeout={agent_cfg.agent_db_connection_timeout.get()}
```

Comment:
This is unexpected?

I would expect the scheduler to use it's own settings, not those of the server?

Reply:
Agreed, cleaned this up and refactored the `server_db_connect` function into `initialize_database_connection_pool` with proper arguments.